src/Type/Php/PhpCoreFunctionsReturnTypeExtension.php

```diff
@@ -0,0 +1,211 @@
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\NullType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+use PHPStan\Type\TypehintHelper;
+
+class PhpCoreFunctionsReturnTypeExtension implements \PHPStan\Type\DynamicFunctionReturnTypeExtension
+{
+
+	/** @var string[] */
+	private $coreFunctions = [
+		'zend_version' => 'string',
+		'func_num_args' => 'int',
+		'func_get_arg' => 'mixed|false',
+		'func_get_args' => 'array',
+		'strlen' => 'int',
+		'strcmp' => 'int',
+		'strncmp' => 'int',
+		'strcasecmp' => 'int',
+		'strncasecmp' => 'int',
+		'each' => 'array|false',
+		'error_reporting' => 'int',
+		'define' => 'bool',
+		'defined' => 'bool',
+		'get_class' => 'string|false',
+		'get_called_class' => 'string|false',
+		'get_parent_class' => 'string|false',
+		'method_exists' => 'bool',
+		'property_exists' => 'bool',
+		'trait_exists' => 'bool',
+		'class_exists' => 'bool',
+		'interface_exists' => 'bool',
+		'function_exists' => 'bool',
+		'class_alias' => 'bool',
+		'get_included_files' => 'string[]',
+		'get_required_files' => 'string[]',
+		'is_subclass_of' => 'bool',
+		'is_a' => 'bool',
+		'get_class_vars' => 'array',
+		'get_object_vars' => 'array',
+		'get_class_methods' => 'string[]',
+		'trigger_error' => 'bool',
+		'user_error' => 'bool',
+		'set_error_handler' => 'mixed',
+		'set_exception_handler' => 'callback',
+		'restore_exception_handler' => 'true',
+		'get_declared_classes' => 'string[]',
+		'get_declared_interfaces' => 'string[]',
+		'get_declared_traits' => 'string[]',
+		'get_defined_functions' => 'string[][]',
+		'get_defined_vars' => 'array',
+		'create_function' => 'string|false',
+		'get_resource_type' => 'string|false',
+		'get_loaded_extensions' => 'string[]',
+		'extension_loaded' => 'bool',
+		'get_extension_funcs' => 'string[]',
+		'get_defined_constants' => 'array',
+		'debug_backtrace' => 'array',
+		'error_clear_last' => 'void',
+		'debug_print_backtrace' => 'void',
+		'gc_collect_cycles' => 'int',
+		'gc_enabled' => 'bool',
+		'gc_enable' => 'void',
+		'gc_disable' => 'void',
+		'sapi_windows_cp_get' => 'int',
+		'sapi_windows_cp_set' => 'bool',
+		'sapi_windows_cp_conv' => 'string',
+		'sapi_windows_cp_is_utf8' => 'bool',
+		'is_iterable' => 'bool',
+	];
+
+	/** @var int[] */
+	private $arrayFunctionsThatCreateArrayBasedOnClosureReturnType = [
+		'array_map' => 0,
+	];
+
+	/** @var int[] */
+	private $arrayFunctionsThatDependOnClosureReturnType = [
+		'array_reduce' => 1,
+	];
+
+	/** @var int[] */
+	private $arrayFunctionsThatDependOnArgumentType = [
+		'array_filter' => 0,
+		'array_unique' => 0,
+		'array_reverse' => 0,
+	];
+
+	/** @var int[] */
+	private $arrayFunctionsThatCreateArrayBasedOnArgumentType = [
+		'array_fill' => 2,
+		'array_fill_keys' => 1,
+	];
+
+	/** @var string[] */
+	private $functionsThatCombineAllArgumentTypes = [
+		'min' => '',
+		'max' => '',
+	];
+
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool
+	{
+		return true;
```

Comment:
There shouldn't be return true because this will override any other registered extension.